capsules/src/process_console.rs

```diff
@@ -170,6 +174,68 @@ impl Default for WriterState {
     }
 }
 
+#[derive(Copy, Clone)]
+pub struct Command<const LEN: usize> {
+    buf: [u8; LEN],
+    len: usize,
+}
+impl<const LEN: usize> Command<LEN> {
+    pub fn new() -> Command<LEN> {
+        Command {
+            buf: [0; LEN],
+            len: 0,
+        }
+    }
+
+    pub fn fill(&mut self, buf: &[u8], terminator_idx: usize) -> Result<(), ErrorCode> {
+        if terminator_idx >= LEN {
+            self.len = LEN;
+        } else {
+            self.len = terminator_idx;
+        }
+
+        for i in 0..self.len {
+            self.buf[i] = buf[i];
+        }
+
+        Ok(())
+    }
+
+    pub fn is_buffer_empty(&mut self) -> Option<bool> {
+        Some(self.len == 0 && self.buf[0] == 0)
+    }
+
+    pub fn is_byte_empty(&mut self, byte_idx: usize) -> Option<bool> {
+        if byte_idx >= self.len {
+            return None;
+        }
+
+        Some(self.buf[byte_idx] == 0)
+    }
+
+    pub fn get_byte(&mut self, byte_idx: usize) -> Option<u8> {
+        if byte_idx >= self.len {
+            return None;
+        }
+
+        Some(self.buf[byte_idx])
+    }
+
+    pub fn same_bytes(&mut self, buf: &[u8], terminator_idx: usize) -> Option<bool> {
```

Comment:
```suggestion
    pub fn same_bytes(&mut self, buf: &[u8; COMMAND_BUF_LEN], terminator_idx: usize) -> Option<bool> {
```

Using a reference to an array instead of a slice will most probably make more efficient code, as the length is known at compile time.

Reply:
Can we use here array instead of a slice, the send slice is command which is a type of &[u8]. In other words I can try adding command.try_into() to transform the slice into an array of [u8; COMMAND_BUF_LEN] but it may panic, wouldn't it be better to handle this situation in the same_bytes method?

Reply:
My suggestion was to use an array instead of a slice. I do not understand your question.

Reply:
The call for the same_bytes is as follows
```RUST
cmd_arr[0].same_bytes(command, terminator + 1);
```

Where the command is a slice and I do not have the guarantee that command.len() == COMMAND_BUF_LEN which may cause a panic at runtime

Reply:
This is why I suggested using arrays instead of slices.

Reply:
Done